### PR TITLE
Np 1387 split database service in two modules 

### PR DESCRIPTION
**Pull request by axthosarouris**
DatabaseServiceImpl contains two "subservices", UserService and RoleService. 

Feature tests were deleted too.

### CODE REVIEW

File: user-access-commons/src/main/java/no/unit/nva/database/DatabaseService.java [hunk 1]
```diff
@@ -17,8 +16,6 @@
 
     List<UserDto> listUsers(String institutionId) throws InvalidEntryInternalException;
 
-    Optional<UserDto> getUserAsOptional(UserDto queryObject) throws InvalidEntryInternalException;
```
Comment (axthosarouris) [inline user-access-commons/src/main/java/no/unit/nva/database/DatabaseService.java:20]: these methods are not used at all.

File: user-access-commons/src/main/java/no/unit/nva/database/DatabaseServiceImpl.java [hunk 2]
```diff
@@ -1,66 +1,33 @@
...
 import org.slf4j.LoggerFactory;
 
 
-//TODO break it up in modules. Coming in next PR.
 @SuppressWarnings("PMD.GodClass")
 public class DatabaseServiceImpl implements DatabaseService {
 
     public static final String DYNAMO_DB_CLIENT_NOT_SET_ERROR = "DynamoDb client has not been set";
 
-    public static final String EMPTY_INPUT_ERROR_MESSAGE = "Expected non-empty input, but input is empty";
```
Comment (axthosarouris) [inline user-access-commons/src/main/java/no/unit/nva/database/DatabaseServiceImpl.java:49]: moved to the subservices

File: user-access-commons/src/main/java/no/unit/nva/database/DatabaseSubService.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Optional;
+import no.unit.nva.database.interfaces.DynamoEntryWithRangeKey;
+import no.unit.nva.exceptions.EmptyInputException;
+import no.unit.nva.exceptions.InvalidEntryInternalException;
+import no.unit.nva.exceptions.InvalidInputException;
+import no.unit.nva.model.JsonSerializable;
+import no.unit.nva.model.Validable;
+import nva.commons.utils.attempt.Failure;
+
+public class DatabaseSubService {
```
Comment (axthosarouris) [inline user-access-commons/src/main/java/no/unit/nva/database/DatabaseSubService.java:17]: common methods for UserService and RoleService

File: user-access-commons/src/main/java/no/unit/nva/database/RoleService.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        logger.debug(ADD_ROLE_DEBUG_MESSAGE + convertToStringOrWriteErrorMessage(roleDto));
+
+        validate(roleDto);
+        checkRoleDoesNotExist(roleDto);
+        table.putItem(roleDto.toRoleDb().toItem());
+    }
+
+    /**
+     * Fethes a role from the database.
```
Comment (brinxmat) [inline user-access-commons/src/main/java/no/unit/nva/database/RoleService.java]: Fetches

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**